versions/latest/modules/en/pages/datastore/backup_restore.adoc

```diff
@@ -150,101 +159,316 @@ The arguments below exist for both the `server` and `etcd-snapshot` subcommands.
+It is possible to restore an etcd snapshot to a different host than it was taken on. When doing so, you must pass the xref:security/token.adoc#_server[server token] that was originally used when taking the snapshot, as it is used to decrypt the bootstrap data inside the snapshot. The process is the same as above but changing step 2 by:
 
-You can manually prune "on-demand" snapshots down to a smaller amount using the following command:
+. In the node that took the snapshot save the value of: `/var/lib/rancher/rke2/server/token`. This is `<BACKED-UP-TOKEN-VALUE>` in step 3.
+. Copy the snapshot to the new node. The path in the node is `<PATH-TO-SNAPSHOT>` in step 3
+. Initiate the restore from snapshot on the first server node with the following commands:
 
 [,bash]
 ----
-rke2 etcd-snapshot prune --etcd-snapshot-retention <NUM-OF-SNAPSHOTS-TO-RETAIN>
+rke2 server \
+  --cluster-reset \
+  --cluster-reset-restore-path=<PATH-TO-SNAPSHOT>
+  --token=<BACKED-UP-TOKEN-VALUE>
 ----
 
-You can manually prune "scheduled" snapshots down to a smaller amount using the following command:
+The token value can also be set in the RKE2 config file.
 
-[,bash]
+[CAUTION]
+====
+
+. Node resources are also included in the etcd snapshot. If restoring to a new set of nodes, you will need to manually delete any old nodes that are no longer present in the cluster.
+. If there is a token set in the RKE2 config file, make sure it is the same as the `<BACKED-UP-TOKEN-VALUE>`, otherwise RKE2 will fail to start.
+====
+
+== ETCDSnapshotFile Custom Resources
+
+[IMPORTANT]
+.Version Gate
+====
+ETCDSnapshotFiles are available as of the November 2023 releases: v1.28.4+rke2r1, v1.27.8+rke2r1, v1.26.11+rke2r1, v1.25.16+rke2r1
+====
+
+Snapshots can be viewed remotely using any Kubernetes client by listing or describing cluster-scoped `ETCDSnapshotFile` resources.
+Unlike the `rke2 etcd-snapshot list` command, which only shows snapshots visible to that node, `ETCDSnapshotFile` resources track all snapshots present on cluster members.
+
+[,shell-session]
 ----
-rke2 etcd-snapshot prune --name etcd-snapshot --etcd-snapshot-retention <NUM-OF-SNAPSHOTS-TO-RETAIN>
+$ kubectl get etcdsnapshotfile
+Name                              Location                                                                           Size     Created
+etcd-snapshot-server-0-1754906881 s3://test-bucket/test-folder/etcd-snapshot-server-0-1754906881                     8937504  2025-08-11T10:08:01Z
+etcd-snapshot-server-0-1754906881 file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-server-0-1754907185 8937504  2025-08-11T10:08:01Z
+etcd-snapshot-server-0-1754907185 s3://test-bucket/test-folder/etcd-snapshot-server-0-1754907185                     9633824  2025-08-11T10:13:05Z
+etcd-snapshot-server-0-1754907185 file:///var/lib/rancher/rke2/server/db/snapshots/etcd-snapshot-server-0-1754907185 9633824  2025-08-11T10:13:05Z
+
+```shell-session
```

Comment:
```suggestion
----

[,shell-session]
```

Not sure if this is one big block or two small separate blocks (which is what my suggestion is based on).